dev_setup.sh

```diff
@@ -621,7 +627,10 @@ if [[ ! -w /var/log/mycroft/ ]] ; then
     if [[ ! -d /var/log/mycroft/ ]] ; then
         $SUDO mkdir /var/log/mycroft/
     fi
-    $SUDO chmod 777 /var/log/mycroft/
+    setup_user=$USER
+    setup_group=$(id -gn "$USER")
+    $SUDO chown -R "${setup_user}":"${setup_group}" /var/log/mycroft
+    $SUDO chmod -R 664 /var/log/mycroft/
```

Comment:
Yeah I assume the intention behind 777 was in the event of multi-user systems. However I'm not sure how common they would really be. 

Perhaps this should be its own PR and anyone interested can have their say.

Reply:
Generally I like to avoid 777 wherever possible...775, as suggested in the open issue, may make sense. But without context of the reasoning behind using 777 I think I'll omit it from the separate PR in favor of addressing it in a more security-focused PR. That's a much larger work effort and probably not one I'd care to undertake without access to more dev environments. Right now I just have Picroft up and running. :)